feat(dev): use PathsMut for debounced RecommendedWatcher

sapphi-red opened the pull request:
Similar to #6120, but for the debounced RecommendedWatcher.

Switched to a different branch: https://github.com/sapphi-red/notify/commits/patches/

sapphi-red commented:
* **#6137** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/6137?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>: 2  dependent PRs ([#6138](https://github.com/rolldown/rolldown/pull/6138) <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/6138?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>, [#6139](https://github.com/rolldown/rolldown/pull/6139) <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/6139?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>) 👈 <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/6137?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add the label <kbd>graphite: merge</kbd> to this PR to add it to the merge queue.

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/rolldown?ref=merge-queue-instructions-in-stack-comment&prId=6137).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

hyf0 commented on crates/rolldown_watcher/src/debounced_recommended_watcher.rs:
```diff
@@ -45,4 +48,9 @@ impl Watcher for DebouncedRecommendedWatcher {
     Ok(())
   }
+
+  fn paths_mut<'me>(&'me mut self) -> Box<dyn crate::PathsMut + 'me> {
```
Could you add it to debounce poll-based watcher? Just in case.

sapphi-red replied:
Sure, I'll add it in a separate PR.

sapphi-red committed:
feat(dev): use PathsMut for debounced RecommendedWatcher (#6137)

Similar to #6120, but for the debounced RecommendedWatcher.

Switched to a different branch: https://github.com/sapphi-red/notify/commits/patches/
